### PR TITLE
Fix for #14208: Order of catalog images not respected on store view level

### PR DESCRIPTION
**Pull request by XxXgeoXxX**
### Description (#14208)
Order of catalog images not respected on store view level

### Preconditions

    Magento 2.2.2

### Linked issues:

magento/community-features#23
Steps to reproduce

    1. Go to product that has multiple images
    2. set scope to all storeviews
    3. change the first image in the gallery
    4. save the product
    5. switch to store view scope (notice that the image order is correct, for now)
    6. save the product
    7. switch back to all store views
    8. change the order of the images again
    9. save the product
    10. go back to storeview scope
    11. notice that the order of the images is not correct anymore.

There should be some sort of 'use default value' checkbox for the media gallery, or at least for the position.
#### Expected result

    The image positions should be according the 'All store views'

#### Actual result

    The image positions use there own position, copied from the 'all storeviews', and don't change anymore according to the 'all store views'

### Fixed Issues
1. magento/magento2#14208: Order of catalog images not respected on store view level

### Manual testing scenarios
Steps to reproduce
   1. Go to product that has multiple images
   2. set scope to all storeviews
   3. change the first image in the gallery
   4. save the product
   5. switch to store view scope (notice that the image order is correct, for now)
   6. save the product
   7. switch back to all store views
   8. change the order of the images again
   9. save the product
   10. go back to storeview scope
   11. notice that the order of the images is not correct anymore.

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by XxXgeoXxX** [comment]
@novikor - please, review issue

File: app/code/Magento/Catalog/Test/Unit/Model/ResourceModel/Product/GalleryTest.php
Comment (orlangur) [inline app/code/Magento/Catalog/Test/Unit/Model/ResourceModel/Product/GalleryTest.php:480]: To test interaction with a database there should be integration test instead.

**Comment by magento-cicd2** [comment]
[CLA assistant check](https://cla.dev.magento.com/magento/magento2?pullRequest=18741) <br/>All committers have signed the CLA.

**Comment by magento-engcom-team** [comment]
Hi @XxXgeoXxX. Thank you for your contribution 
Here is some useful tips how you can test your changes using Magento test environment.
Add the comment under your pull request to deploy test or vanilla Magento instance:
- `@magento-engcom-team give me test instance` - deploy test instance based on PR changes
- `@magento-engcom-team give me $VERSION instance` - deploy vanilla Magento instance

For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation

**Comment by sidolov** [comment]
Hi @XxXgeoXxX , looks like you made some commits with email different than in your GitHub profile, please, add email from commits to your profile!
Thank you!

**Comment by XxXgeoXxX** [comment]
Hi, @sidolov, I updated local settings for GIT. Unfortunately, email which was added in commits is locally and I couldn't verify it.

**Comment by XxXgeoXxX** [comment]
Hi, @p-bystritsky
Could you, please, merge this pull request?

**Comment by josefbehr** [comment]
@XxXgeoXxX could you please first resolve the review?

**Comment by XxXgeoXxX** [comment]
Hi, @josefbehr
Could you accepted this PR without integration test(unit and manual tests were pass) or it required?

**Comment by josefbehr** [comment]
@XxXgeoXxX would be better to have. do you need help with it?

File: app/code/Magento/Catalog/Test/Unit/Model/ResourceModel/Product/GalleryTest.php
Comment (josefbehr) [reply app/code/Magento/Catalog/Test/Unit/Model/ResourceModel/Product/GalleryTest.php:480]: @orlangur can you specify for @XxXgeoXxX what you'd expect, or possibly even help implement it?

Comment (josefbehr) [reply app/code/Magento/Catalog/Test/Unit/Model/ResourceModel/Product/GalleryTest.php:480]: @orlangur I'll accept it for know, as it's the same as the accepted PR for 2.3 branch. If you can specify the needed test, please do to, and @XxXgeoXxX can add it as a new PR.

**Review by josefbehr — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @josefbehr, thank you for the review.
[ENGCOM-4375](https://jira.corp.magento.com/browse/ENGCOM-4375) has been created to process this Pull Request

**Comment by ihor-sviziev** [comment]
Hi @XxXgeoXxX, @josefbehr,

It looks like we have better solution for removing values on store view levels:
https://github.com/magento/magento2/pull/21041

**Comment by XxXgeoXxX** [comment]
Hi @ihor-sviziev
I use Magento logic from configuration "Use Default value". This functionality removes values per stores/websites and left only default. That means you have less invaluable values in database and data in the database has less size. As I looked in #21041 we collect a lot of data and sort it on the frontend.

**Comment by VasylShvorak** [comment]
On hold until [#18778](https://github.com/magento/magento2/pull/18778) would be accepted

**Comment by sidolov** [comment]
@XxXgeoXxX I'm closing this PR because PR with the same changes to 2.3-develop is closed due to inactivity, please, reopen PR https://github.com/magento/magento2/pull/18778 if you wish to continue. 
Thank you for your involvement!

**Pull request closed without merging**
